cpu/lm4f120/periph/spi.c

```diff
@@ -0,0 +1,273 @@
+        unsigned long rx;
+        unsigned long tx;
+        unsigned long mask;
+    } pins;
+};
+
+static const struct spi_conf_s spi_confs[] = {
+#if SPI_0_EN
+    [SPI_0] = {
+        .ssi_sysctl = SYSCTL_PERIPH_SSI0,
+        .ssi_base = SSI0_BASE,
+        .gpio_sysctl = SYSCTL_PERIPH_GPIOA,
+        .gpio_port = GPIO_PORTA_BASE,
+        .pins = {
+            .clk = GPIO_PA2_SSI0CLK,
+            .fss = GPIO_PA3_SSI0FSS,
+            .rx = GPIO_PA4_SSI0RX,
+            .tx = GPIO_PA5_SSI0TX,
+            .mask = GPIO_PIN_2 | GPIO_PIN_3 | GPIO_PIN_4 | GPIO_PIN_5
+        }
+    },
+#endif
+
+#if SPI_1_EN
+    [SPI_1] = {
+        .ssi_sysctl = SYSCTL_PERIPH_SSI1,
+        .ssi_base = SSI1_BASE,
+        .gpio_sysctl = SYSCTL_PERIPH_GPIOF,
+        .gpio_port = GPIO_PORTF_BASE,
+        .pins = {
+            .clk = GPIO_PF2_SSI1CLK,
+            .fss = GPIO_PF3_SSI1FSS,
+            .rx = GPIO_PF0_SSI1RX,
+            .tx = GPIO_PF1_SSI1TX,
+            .mask = GPIO_PIN_0 | GPIO_PIN_1 | GPIO_PIN_2 | GPIO_PIN_3
+        }
+    },
+#endif
+
+#if SPI_2_EN
+    [SPI_2] = {
+        .ssi_sysctl = SYSCTL_PERIPH_SSI2,
+        .ssi_base = SSI2_BASE,
+        .gpio_sysctl = SYSCTL_PERIPH_GPIOB,
+        .gpio_port = GPIO_PORTB_BASE,
+        .pins = {
+            .clk = GPIO_PB4_SSI2CLK,
+            .fss = GPIO_PB5_SSI2FSS,
+            .rx = GPIO_PB6_SSI2RX,
+            .tx = GPIO_PB7_SSI2TX,
+            .mask = GPIO_PIN_4 | GPIO_PIN_5 | GPIO_PIN_6 | GPIO_PIN_7
+        }
+    },
+#endif
+
+#if SPI_3_EN
+    [SPI_3] = {
+        .ssi_sysctl = SYSCTL_PERIPH_SSI3,
+        .ssi_base = SSI3_BASE,
+        .gpio_sysctl = SYSCTL_PERIPH_GPIOD,
+        .gpio_port = GPIO_PORTD_BASE,
+        .pins = {
+            .clk = GPIO_PD0_SSI3CLK,
+            .fss = GPIO_PD1_SSI3FSS,
+            .rx = GPIO_PD2_SSI3RX,
+            .tx = GPIO_PD3_SSI3TX,
+            .mask = GPIO_PIN_0 | GPIO_PIN_1 | GPIO_PIN_2 | GPIO_PIN_3
+        }
+    },
+#endif
+};
+
+int spi_init_master(spi_t dev, spi_conf_t conf, spi_speed_t speed)
+{
+    unsigned long initialData = 0;
+    unsigned long mode = 0;
+    unsigned long bitrate;
+
+    if (dev >= SPI_NUMOF) {
+        return -1;
+    }
+
+    spi_poweron(dev);
+    ROM_SSIDisable(spi_confs[dev].ssi_base);
+    ROM_SysCtlPeripheralEnable(spi_confs[dev].gpio_sysctl);
+
+    ROM_GPIOPinConfigure(spi_confs[dev].pins.clk);
+    ROM_GPIOPinConfigure(spi_confs[dev].pins.fss);
+    ROM_GPIOPinConfigure(spi_confs[dev].pins.rx);
+    ROM_GPIOPinConfigure(spi_confs[dev].pins.tx);
+
+    ROM_GPIOPinTypeSSI(spi_confs[dev].gpio_port, spi_confs[dev].pins.mask);
+
+    switch (speed) {
+        case SPI_SPEED_100KHZ:
+            bitrate = 100000;
+            break;
+        case SPI_SPEED_1MHZ:
+            bitrate = 1000000;
+            break;
+        case SPI_SPEED_5MHZ:
+            bitrate = 5000000;
+            break;
+        case SPI_SPEED_10MHZ:
+            bitrate = 10000000;
+            break;
+        default:
```

Comment:
Whatever you prefer. I tend to prefer to see all enum values listed in a switch case so that there is no question whether the missing case has been forgotten or not.